### ISSUE TITLE
fd_write not respecting offset from fd_seek

### ISSUE THREAD

**Issue by mischnic** [wasmerio/wasmer-js#165]
The `fd_write` implementation doesn't seem to respect an offset previously set by fseek, instead it always writes to the end.

#### Reproduction

```c
#include <stdio.h>

const char *source = "/absolute/path/to/fseek_test";

int main(void) {
    long offset = 4;

    FILE *f = fopen(source, "w");
    if(!f){
        perror("fopen");
    }
    if(fprintf(f, "%s", "123456789123456789") == 0){
        perror("fprintf");
    }

    if(fseek(f, 4, SEEK_SET) != 0){
        perror("fseek");
    }
    if(fprintf(f, "%s", "abc") == 0){
        perror("fprintf");
    }
    if(fseek(f, 0, SEEK_END) != 0){
        perror("fseek");
    }
    if(fprintf(f, "%s", "end") == 0){
        perror("fprintf");
    }

    if(fclose(f) != 0){
        perror("fclose");
    }
    return 0;
}
```
(with https://github.com/CraneStation/wasi-sdk/releases/tag/wasi-sdk-8)

```sh
# compile
clang --target=wasm32-unknown-wasi --sysroot ..../wasi-libc/sysroot test.c

# run, produces `123456789123456789abcend`
wasmer-js run a.out --dir=/

# run, produces `1234abc89123456789end` (correct, same as with "real" libc)
wasmtime --dir=/ a.out
```

**Comment by torch2424** [comment]
@mischnic Thank you soooo much for such a detailed issue with a repro case!

cc @syrusakbary as they have been doing a lot of the Wasi polyfill work here.

If they don't get the chance to get to it, I'll try to when I get the chance! 😄

**Comment by syrusakbary** [comment]
Hi @mischnic both issues should now be fixed on master.

**Comment by syrusakbary** [comment]
We just published a new version of wasmer-js `0.5.1` that should have this issue fixed.
